src/test/consensus/Consensus_test.cpp

```diff
@@ -522,6 +522,162 @@ class Consensus_test : public beast::unit_test::suite
+
+            Sim sim(
+                parms,
+                tg,
+                topology(tg, [&](PeerID i, PeerID j) {
+                        auto delayFactor = (i <= 1 || j <= 1) ? 1.1 : 0.2;
+                        return round<milliseconds>(
+                            delayFactor * parms.ledgerGRANULARITY);
+                    }));
+
+            // Run to the ledger *prior* to decreasing the resolution
+            sim.run(increaseLedgerTimeResolutionEvery - 2);
+
+            // In order to create the discrepency, we want a case where if
+            //   X = effCloseTime(closeTime, resolution, parentCloseTime)
+            //   X != effCloseTime(X, resolution, parentCloseTime)
+            //
+            // That is, the effective close time is not a fixed point. This can
+            // happen if X = parentCloseTime + 1, but a subsequent rounding goes
+            // to the next highest multiple of resolution.
+
+            // So we want to find an offset  (now + offset) % 30s = 15
+            //                               (now + offset) % 20s = 15
+            // This way, the next ledger will close and round up   Due to the
+            // network delay settings, the round of consensus will take 5s, so
+            // the next ledger's close time will
+
+            NetClock::duration when = sim.peers[0].now().time_since_epoch();
+
+            // Check we are before the 30s to 20s transition
+            NetClock::duration resolution =
+                sim.peers[0].lastClosedLedger.closeTimeResolution();
+            BEAST_EXPECT(resolution == NetClock::duration{30s});
+
+            while (
+                ((when % NetClock::duration{30s}) != NetClock::duration{15s}) ||
+                ((when % NetClock::duration{20s}) != NetClock::duration{15s}))
+                when += 1s;
+            // Advance the clock without consensus running (IS THIS WHAT
+            // PREVENTS IT IN PRACTICE?)
+            sim.net.step_for(
+                NetClock::time_point{when} - sim.peers[0].now());
+
+            // Run one more ledger with 30s resolution
+            sim.run(1);
+
+            // close time should be ahead of clock time since we engineered
+            // the close time to round up
+            for (Peer const & peer : sim.peers)
+            {
+                BEAST_EXPECT(
+                    peer.lastClosedLedger.closeTime() > peer.now());
+                BEAST_EXPECT(peer.lastClosedLedger.closeAgree());
+                BEAST_EXPECT(
+                    peer.lastClosedLedger.id() ==
+                    sim.peers[0].lastClosedLedger.id());
+            }
+
+
+            // All peers submit their own ID as a transaction
+            for (Peer & peer : sim.peers)
+                peer.submit(Tx{static_cast<std::uint32_t>(peer.id)});
+
+            // Run 1 more round, this time it will have a decreased
+            // resolution of 20 seconds.
+            // 86481
```

Comment:
although I see this value mentioned as raw close time in the following comments, I don't think it has any context in this particular comment  - maybe remove?